instana/instrumentation/sqlalchemy.py

```diff
@@ -47,23 +50,43 @@ def receive_after_cursor_execute(**kw):
             if scope is not None:
                 scope.close()
 
-    @event.listens_for(Engine, 'dbapi_error', named=True)
-    def receive_dbapi_error(**kw):
-        context = kw['context']
 
-        if context is not None and hasattr(context, '_stan_scope'):
-            scope = context._stan_scope
-            if scope is not None:
-                scope.span.mark_as_errored()
+    error_event = "handle_error"
+    # Handle dbapi_error event; deprecated since version 0.9
+    if sqlalchemy.__version__[0] == "0":
+        error_event = "dbapi_error"
 
-                if 'exception' in kw:
-                    e = kw['exception']
-                    scope.span.set_tag('sqlalchemy.err', str(e))
-                else:
-                    scope.span.set_tag('sqlalchemy.err', "No dbapi error specified.")
-                scope.close()
 
+    def _set_error_tags(scope, exception):
+        scope.span.mark_as_errored()
+        if exception:
+            scope.span.set_tag('sqlalchemy.err', str(exception))
+        else:
+            scope.span.set_tag('sqlalchemy.err', "No %s specified." % error_event)
+        scope.close()
+
+
+    @event.listens_for(Engine, error_event, named=True)
+    def receive_handle_db_error(**kw):
+
+        # support older db error event
+        if error_event == "dbapi_error":
+            context = kw.get('context')
+            if hasattr(context, '_stan_scope') and \
+                    hasattr(context, 'exception'):
+                scope = context._stan_scope
+                context_exception = context.exception
+                if scope:
+                    _set_error_tags(scope, context_exception)
+
+        else:
+            context = kw.get('exception_context')
+            if hasattr(context.execution_context, '_stan_scope') and \
+                    hasattr(context, 'sqlalchemy_exception'):
+                scope = context.execution_context._stan_scope
+                context_exception = context.sqlalchemy_exception
+                if scope:
+                    _set_error_tags(scope, context_exception)
 
-    logger.debug("Instrumenting sqlalchemy")
```

Comment:
The idea for this log line to be the last is that we write it once instrumentation has succeeded. Should there be an exception during initialization, there will be no such line in the log, which gives us a hint that there was a problem.